### PR TITLE
1743 junit authorisations

### PR DESCRIPTION
**Pull request by ConorH994**
https://tools.hmcts.net/jira/browse/AM-1743

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJrdOfficeOrgRoleMappingTest.java [hunk 1]
```diff
@@ -159,6 +161,46 @@ void shouldReturnTribunalJudgeSalariedRoles_withIAC() {
...
 
+    @Test
+    @DisplayName("Scenario 4: IAC Judge with Non IAC Auth")
+    void iacJudgeWithoutIacAuth() {
+
+        judicialAccessProfiles.forEach(judicialAccessProfile -> {
+            judicialAccessProfile.setAppointment("Tribunal Judge");
+            judicialAccessProfile.setAppointmentType("Salaried");
+            judicialAccessProfile.setServiceCode("BFA1");
+            judicialAccessProfile.setTicketCodes(List.of("35970"));
```
Comment (sankar-padakula) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJrdOfficeOrgRoleMappingTest.java:172]: Can you please change this to 373. BFA1's ticket code is 373 and all ticket codes will be 3 digits

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJrdOfficeOrgRoleMappingTest.java [hunk 2]
```diff
@@ -628,4 +681,539 @@ void shouldReturnRoles_withPresidentAppointment_withResidentRole_withIAC() {
...
+    @Test
+    @DisplayName("Scenario 2: Missing Service code in Appointment")
+    void shouldCreateAssignmentsWhenAppointmentServiceCodeIsMissing() {
+
+        judicialAccessProfiles.forEach(judicialAccessProfile -> {
+            judicialAccessProfile.setAppointment("Employment Judge");
+            judicialAccessProfile.setAppointmentType("fee paid");
+            judicialAccessProfile.setServiceCode(null);
+            judicialAccessProfile.setAuthorisations(
+                    List.of(Authorisation.builder().userId(UUID.randomUUID().toString()).serviceCode("BFA1").build(),
```
Comment (sankar-padakula) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJrdOfficeOrgRoleMappingTest.java:694]: userId is not required and its planned to remove from Authorisation. So please remove .userId(UUID.randomUUID().toString())

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJrdOfficeOrgRoleMappingTest.java [hunk 3]
```diff
@@ -628,4 +681,539 @@ void shouldReturnRoles_withPresidentAppointment_withResidentRole_withIAC() {
...
+            judicialAccessProfile.setAppointmentType("fee paid");
+            judicialAccessProfile.setServiceCode(null);
+            judicialAccessProfile.setTicketCodes(List.of("A", "B", "C", "D"));
+        });
+
+        JudicialAccessProfile profile = TestDataBuilder.buildJudicialAccessProfile();
+        profile.setAppointment("Tribunal Judge");
+        profile.setAppointmentType("SPTW");
+        profile.setServiceCode(null);
+        profile.setTicketCodes(List.of("A", "B", "C", "D"));
```
Comment (sankar-padakula) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJrdOfficeOrgRoleMappingTest.java:841]: Ticket codes will be 3 digit codes. please change accordingly

**Review by sankar-padakula — COMMENTED** [review]
Minor changes. Please correct it

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJrdOfficeOrgRoleMappingTest.java (hunk 1 above)
Comment (ConorH994) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJrdOfficeOrgRoleMappingTest.java:172]: ok will do

Comment (ConorH994) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJrdOfficeOrgRoleMappingTest.java:172]: this was in the excel sheet scenarios to be honest

but i will change it to suit

**Review by sankar-padakula — APPROVED** [review]

**Review by sankar-padakula — APPROVED** [review]

**Review by aashishaggarwal123 — APPROVED** [review]

**Review by nitinsahu1 — APPROVED** [review]

**Pull request merged**